src/operator/bufferToggle.ts

```diff
@@ -44,18 +46,18 @@ import {errorObject} from '../util/errorObject';
  * @owner Observable
  */
 export function bufferToggle<T, O>(openings: Observable<O>,
```

Comment:
Yes, anywhere we're accepting an `Observable` we should accept any of the acceptable types. Although synchronous things like iterators and Arrays don't make sense here. I think we should probably accept `Promise` though.

Basically, we need to accept interfaces of `Promise`, `Subscribable` and `IObservable` here. (We probably also need better names for those last two, haha)